erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -29,6 +29,8 @@ def validate(self):
 	def on_submit(self):
 		self.update_status()
 		self.update_delivery_notes()
+		if not self.driver:
+			frappe.throw(_("A driver must be set to submit."))
```

Comment:
@blaggacao move this validation to `validate()` and use self._action == "submit"

Reply:
@s-aga-r Thanks for the hint, addressed in this [diff](https://github.com/frappe/erpnext/compare/228084bac86f571ac1e36b8f4fe26ed0215fe2ee..5ea4c473346391cdc9268c97df8739d8d5ce26bb)